python/lsst/daf/butler/registry/interfaces/_database.py

```diff
@@ -263,6 +265,31 @@ def dropTemporaryTable(self, table: sqlalchemy.schema.Table) -> None:
         else:
             raise TypeError(f"Table {table.key} was not created by makeTemporaryTable.")
 
+    @contextmanager
+    def temporary_table(self, spec: ddl.TableSpec, name: Optional[str]) -> Iterator[sqlalchemy.schema.Table]:
+        """Return a context manager that creates and then drops a context
+        manager.
+
+        Parameters
+        ----------
+        spec : `ddl.TableSpec`
+            Specification for the columns.  Unique and foreign key constraints
+            may be ignored.
+        name : `str`, optional
+            If provided, the name of the SQL construct.  If not provided, an
+            opaque but unique identifier is generated.
+
+        Returns
+        -------
+        table : `sqlalchemy.schema.Table`
+            SQLAlchemy representation of the table.
+        """
+        table = self.makeTemporaryTable(spec=spec, name=name)
+        try:
+            yield table
+        finally:
+            self.dropTemporaryTable(table)
```

Comment:
I think Coverity tries to say that there is no unit test for the whole `temporary_table` but somehow it only points to a single line.